### PR TITLE
feat(xo): type definition for v0.26

### PR DESCRIPTION
**Pull request by peterblazejewicz**
- definition file
- tests

https://github.com/xojs/xo/#config

Thanks!

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. 
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/xo/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+export {};
+/**
+ * From T pick a set of properties K
+ */
+type Pick<T, K extends keyof T> = {
+    [P in K]: T[P];
+};
+
+export type getErrorResults = typeof eslint.CLIEngine.getErrorResults;
+export type outputFixes = typeof eslint.CLIEngine.outputFixes;
```
Comment (PranavSenthilnathan) [inline types/xo/index.d.ts]: I don't think this is a type that's exported.
```suggestion
export const outputFixes : typeof eslint.CLIEngine.outputFixes;
```

**Review by PranavSenthilnathan — CHANGES_REQUESTED** [review]

File: types/xo/index.d.ts (hunk 1 above)
Comment (peterblazejewicz) [reply types/xo/index.d.ts]: Yes, very good catch, I've overlooked these two as beign import from ESLint, sorry. 
The definition is rewritten to match shapes and I've added sample usage tests too.
Thanks!

File: types/xo/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
+// Type definitions for xo 0.26
+// Project: https://github.com/xojs/xo#readme
+// Definitions by: Piotr Błażejewicz (Peter Blazejewicz) <https://github.com/peterblazejewicz>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import eslint = require('eslint');
+
+export {};
```
Comment (sheetalkamat) [inline types/xo/index.d.ts]: Can you please remove this

File: types/xo/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+/**
+ * From T pick a set of properties K
+ */
+type Pick<T, K extends keyof T> = {
+    [P in K]: T[P];
+};
+
+// eslint.CLIEngine.getErrorResults without modification
+// eslint.CLIEngine.getErrorResults redeclared to match input shape
+
```
Comment (sheetalkamat) [inline types/xo/index.d.ts]: getFormatter missing?

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/xo/index.d.ts (hunk 2 above)
Comment (peterblazejewicz) [reply types/xo/index.d.ts]: redone and amended. Thx!

File: types/xo/index.d.ts (hunk 3 above)
Comment (peterblazejewicz) [reply types/xo/index.d.ts]: redone/amended

**Pull request merged**